app/models/miq_request_workflow.rb

```diff
@@ -856,19 +856,25 @@ def ems_has_clusters?
   end
 
   def get_ems_folders(folder, dh = {}, full_path = "")
+    path = full_path
     if folder.evm_object_class == :EmsFolder
       if folder.hidden
         return dh if folder.name != 'vm'
       else
-        full_path += full_path.blank? ? folder.name.to_s : " / #{folder.name}"
-        dh[folder.id] = full_path unless folder.type == "Datacenter"
+        path = full_path.dup
+        if path.blank?
+          path << folder.name.to_s
+        else
+          path << " / ".freeze << folder.name
+        end
+        dh[folder.id] = path unless folder.type == "Datacenter".freeze
```

Comment:
> (not sure if you want to pull out into a const)

Per @syncrou 's previous review, he thought it was better as it not being a constant.  I already had it as a constant originally.

> would it make sense to have `elseif folder.type != "Datacenter".freeze ?

Possibly, though I was mostly trying to keep the logic the same, and just adjusting things to limit object allocations.  If you look at the previous code, the `if` logic is basically exactly the same, I just expanded out the ternary operator since I thought it was easier to read in that form.  Figured trying to tweak the logic would add unneeded risk.

Reply:
aah - you are actually using path below